### PR TITLE
remove big lock in arch_phy_irq

### PR DESCRIPTION
**Pull request by hujun260**
## Summary

remove big lock in arch_phy_irq
reason:
We would like to replace the big lock with a small lock.

## Impact
arch_phy_irq

## Testing
ci

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR does not fully meet the NuttX requirements.  While it provides a basic summary, it lacks crucial details.

Here's a breakdown of what's missing:

* **Insufficient Summary:**  "remove big lock in arch_phy_irq" is too vague.  What big lock? What is `arch_phy_irq` and why does it need a lock?  How does the new smaller lock work?  What problem does this solve?  This section needs to be significantly more descriptive.
* **Missing Issue References:** Are there related NuttX or NuttX Apps issues? If so, link them.
* **Impact Section Incomplete:** The impact section just repeats "arch_phy_irq."  Address *all* the impact points (user, build, hardware, documentation, security, compatibility) with "YES" or "NO" and provide descriptions for any "YES" answers.  Simply stating the affected function isn't enough.  For example, does removing this lock have performance implications?  Does it change the API?  Does it affect any specific architectures or boards?
* **Inadequate Testing:** "ci" is not sufficient testing information.  List the specific build hosts and targets used for testing (OS, architecture, board, configuration).  Provide *actual* testing logs demonstrating the problem before the change and the improved behavior after the change. "ci" simply indicates that continuous integration tests were run, not what those tests were or what their results were.

In short, the PR needs much more detail and evidence to demonstrate that the change is correct and well-understood.

File: boards/arm/imxrt/imxrt1050-evk/src/imxrt_ethernet.c [hunk 1]
```diff
@@ -75,6 +75,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/imxrt/imxrt1050-evk/src/imxrt_ethernet.c:82]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/imxrt/imxrt1020-evk/src/imxrt_ethernet.c [hunk 2]
```diff
@@ -72,6 +72,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/imxrt/imxrt1020-evk/src/imxrt_ethernet.c:79]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/stm32/stm32f4discovery/src/stm32_ethernet.c [hunk 3]
```diff
@@ -80,6 +81,8 @@
  * Private Data
  ****************************************************************************/
 
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/stm32/stm32f4discovery/src/stm32_ethernet.c:84]:
```suggestion
static spinlock_t g_ethmac_lock = SP_UNLOCKED;
```

File: boards/arm/samv7/samv71-xult/src/sam_ethernet.c [hunk 4]
```diff
@@ -77,6 +78,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/samv7/samv71-xult/src/sam_ethernet.c:85]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/samv7/same70-xplained/src/sam_ethernet.c [hunk 5]
```diff
@@ -77,6 +78,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/samv7/same70-xplained/src/sam_ethernet.c:85]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/sama5/sama5d4-ek/src/sam_ethernet.c [hunk 6]
```diff
@@ -82,6 +83,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/sama5/sama5d4-ek/src/sam_ethernet.c:90]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/sama5/sama5d3x-ek/src/sam_ethernet.c [hunk 7]
```diff
@@ -81,6 +82,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/sama5/sama5d3x-ek/src/sam_ethernet.c:89]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/sama5/sama5d3-xplained/src/sam_ethernet.c [hunk 8]
```diff
@@ -81,6 +82,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/sama5/sama5d3-xplained/src/sam_ethernet.c:89]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/sama5/sama5d2-xult/src/sam_ethernet.c [hunk 9]
```diff
@@ -81,6 +82,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/sama5/sama5d2-xult/src/sam_ethernet.c:89]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/sama5/jupiter-nano/src/sam_ethernet.c [hunk 10]
```diff
@@ -81,6 +82,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/sama5/jupiter-nano/src/sam_ethernet.c:89]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/sam34/sam4e-ek/src/sam_ethernet.c [hunk 11]
```diff
@@ -71,6 +72,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/sam34/sam4e-ek/src/sam_ethernet.c:79]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/imxrt/teensy-4.x/src/imxrt_ethernet.c [hunk 12]
```diff
@@ -73,6 +73,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/imxrt/teensy-4.x/src/imxrt_ethernet.c:80]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/imxrt/imxrt1170-evk/src/imxrt_ethernet.c [hunk 13]
```diff
@@ -74,6 +74,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/imxrt/imxrt1170-evk/src/imxrt_ethernet.c:81]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/imxrt/imxrt1064-evk/src/imxrt_ethernet.c [hunk 14]
```diff
@@ -73,6 +73,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/imxrt/imxrt1064-evk/src/imxrt_ethernet.c:80]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

File: boards/arm/imxrt/imxrt1060-evk/src/imxrt_ethernet.c [hunk 15]
```diff
@@ -73,6 +73,12 @@
 #  define phyinfo(x...)
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static spinlock_t g_ethernet_lock = SP_UNLOCKED;
```
Comment (xiaoxiang781216) [inline boards/arm/imxrt/imxrt1060-evk/src/imxrt_ethernet.c:80]:
```suggestion
static spinlock_t g_phy_lock = SP_UNLOCKED;
```

Comment (hujun260) [reply boards/arm/imxrt/imxrt1060-evk/src/imxrt_ethernet.c]: done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**